src/lib/datatypes/include/sol-str-slice.h

```diff
@@ -352,6 +352,28 @@ sol_str_slice_trim(struct sol_str_slice slice)
 struct sol_vector sol_str_slice_split(const struct sol_str_slice slice, const char *delim, size_t maxsplit);
 
 /**
+ * @brief Do an one step split iteration over a slice.
+ *
+ * Usage example:
+ * @code{.c}
+ * struct sol_str_slice slice = SOL_STR_SLICE_LITERAL("one;two;three");
+ * struct sol_str_slice token;
+ * const char *itr = NULL;
+ * while (sol_str_slice_split_iterate(slice, &token, &itr, ";")) {
+ *    printf("%.*s\t", SOL_STR_SLICE_PRINT(token));
+ *    //It will print: one two three
+ * }
+ * @endcode
+ *
+ * @param slice The slice the be splitted
+ * @param token A splitted token
+ * @param itr An iterator - It should be @c NULL on the first call.
+ * @param delim The delimiter string
+ * @return @c true if iteration should continue or @c false if iteration should be stopped.
+ */
+bool sol_str_slice_split_iterate(const struct sol_str_slice slice, struct sol_str_slice *token, const char **itr, const char *delim);
```

Comment:
I like delim being a string, should we change the other split to use it?

Also cross-link them with `@see`.